eZ/Publish/API/Repository/TransactionService.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Publish\API\Repository;
+
+use Closure;
+
+interface TransactionService
+{
+    /**
+     * Begin transaction.
+     *
+     * Begins an transaction, make sure you'll call commit or rollback when done,
+     * otherwise work will be lost.
+     */
+    public function beginTransaction(): void;
+
+    /**
+     * Commit transaction.
+     *
+     * Commit transaction, or throw exceptions if no transactions has been started.
+     *
+     * @throws \RuntimeException If no transaction has been started
+     */
+    public function commit(): void;
+
+    /**
+     * Rollback transaction.
+     *
+     * Rollback transaction, or throw exceptions if no transactions has been started.
+     *
+     * @throws \RuntimeException If no transaction has been started
+     */
+    public function rollback(): void;
+
+    /**
+     * Executes a function in a transaction.
+     *
+     * If an exception occurs during execution of the function or transaction commit,
+     * the transaction is rolled back and the exception re-thrown.
+     *
+     * @throws \Throwable
+     *
+     * @return mixed The value returned by $func
+     */
+    public function transactional(Closure $func);
```

Comment:
How about 
```suggestion
    public function transactional(callable $func);
```

Reply:
Fixed in https://github.com/ezsystems/ezplatform-kernel/pull/164/commits/ffb35ff9c0bd6d94b14a1401cfb71a4e9312cb44